### PR TITLE
6664 new datetimepicker

### LINKED ISSUE #6664: As an enterprise manager, I can use a new date picker
## Description
**- As a:** Enterprise user and super admin
**- On pages:** 
Our date picker are used in the following pages:

- `/admin/order_cycles` on ordercycle list but also when creating / editing an order cycle
- `/admin/reports` on each reports filters
- `/admin/orders?q[s]=completed_at+desc` in start / end date filters
- `/admin/orders/bulk_management` in start / end date filters
- `/admin/subscriptions/` when creating or editing a new subscription

**- I want to be able to do:** use a date picker that is not using `jquery-ui-rails` as a dependency

This new date picker must:

✅ Select both date and time
✅ Is already or can be translated
✅ Allow the option in the future to add custom shortcuts such “next month” (as discussed [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/5032#issuecomment-625757174))
✔️ Currently the date picker has the option to select NOW/TODAY. It could be nice to keep it.
✔️ Only if super easy with the new one, it would be nice to improve the selection of the end date. Currently let's say I want to select a time between January the 21st to January the 22nd I have to select:
`2021-01-21 00:00` to `2021-01-23 00:00`.
Selecting the **23**rd ensures that the 22nd is fully selected. So for the end date it's important that if I select the 22 date, it actually does cover the 22nd completely automatically.
However for the start date, the current behavior (starting at midnight) is correct.

Legend: 
✅ : Must have
✔️ : Nice to have

## Acceptance Criteria & Tests

See description for must have criterias, plus:

_Scenario 1 As a simple producer_

We need to make sure selecting dates ends up with correct results on all the following pages:

- `/admin/order_cycles` on ordercycle list but also when creating / editing an order cycle
- `/admin/reports` on each reports filters
- `/admin/orders?q[s]=completed_at+desc` in start / end date filters
- `/admin/orders/bulk_management` in start / end date filters
- `/admin/subscriptions/` when creating or editing a new subscription

_Scenario 2 As a multi-producer hub_

Repeat same tests as scenario 1

_Scenario 3 As a super admin_

Repeat same tests as scenario 2

## Previous discussions

https://github.com/openfoodfoundation/openfoodnetwork/pull/6482

Especially this comment has all the history: https://github.com/openfoodfoundation/openfoodnetwork/pull/6482#issuecomment-758102380

## Todo
- [x] Implements datetimepicker at each location where it is used
- [x] Remove every reference to jQueryUI Datetimepicker
- [x] Transforme each "start date - end date" to a Daterangepicker (see example: https://flatpickr.js.org/examples/#range-calendar ; and comment https://github.com/openfoodfoundation/openfoodnetwork/issues/6664#issuecomment-760964353) --> opened a new issue : #6730
- [x] Add shortcut such as "Now" or "next month"
- [x] Apply small css modifications (see comment https://github.com/openfoodfoundation/openfoodnetwork/issues/6664#issuecomment-760369452) if it's easy. If not, open new issue to customize widget.

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #6664 and #3894
The aim of this PR is to use a new datepicker and remove all the references to the old one (jquery-ui datetimepicker)

#### What should we test?
Test the date(time)picker where it is used. I noticed:
 - `/admin/order_cycles/`
 - `/admin/order_cycles/[ORDER_ID]/edit`
 - `/admin/reports` such as `/order_management/reports/bulk_coop/new` or `admin/reports/orders_and_fulfillment?report_type=order_cycle_customer_totals`
 - `/admin/orders/bulk_management`
 - `/admin/orders?q[s]=completed_at+desc#`

#### Release notes
New date picker widget to select date and time across the product.

Changelog Category: User facing changes

#### What it looks like 
Capture d’écran 2021-01-26 à 21 09 08

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Great! :heart:

I'll take a look at this soon. Usually on PRs that effect something visually we post some screenshots or a GIF in the PR so reviewers can see what it looks like :+1:

**Comment by jibees** [comment]
@Matt-Yorkley Ok, thanks for the advice! I've added a screenshot of the date picker.

File: config/locales/en.yml [hunk 1]
```diff
@@ -3586,8 +3582,10 @@ See the %{link} to find out more about %{sitename}'s features and to start using
           or_enter_new_card: "Or, enter details for a new card:"
           remember_this_card: Remember this card?
     date_picker:
-      format: ! '%Y-%m-%d'
-      js_format: 'yy-mm-dd'
+      flatpickr_date_format: "Y-m-d"
+      flatpickr_datetime_format: "Y-m-d H:i"
+      today: "Today"
+      now: "Now"
```
Comment (sauloperez) [inline config/locales/en.yml:3588]: We'll need to make sure all instances are aware of this change and it gets translated before deploying.

File: spec/features/admin/bulk_order_management_spec.rb [hunk 2]
```diff
@@ -469,17 +469,17 @@
         expect(page).to have_no_selector "tr#li_#{li4.id}"
       end
 
-      it "displays only line items whose orders meet the date restriction criteria, when changed" do
+      it "displays only line items whose orders meet the date restriction criteria, when changed", :debug do
```
Comment (sauloperez) [inline spec/features/admin/bulk_order_management_spec.rb:472]: what's that `:debug` argument?

File: spec/features/admin/order_cycles/simple_spec.rb [hunk 3]
```diff
@@ -12,53 +12,58 @@
...
-      find("input#oc#{oc1.id}_name").set "Updated Order Cycle 1"
-      find("input#oc#{oc1.id}_orders_open_at").set "2040-12-01 12:00:00"
-      find("input#oc#{oc1.id}_orders_close_at").set "2040-12-01 12:00:01"
+    ## -- OC1
+    # Manually fill out name
+    find("input#oc#{oc1.id}_name").set ""
+    fill_in("oc#{oc1.id}_name", :with => "Updated Order Cycle 1")
+    
+    ## -- OC2
+    # Manually fill out name
```
Comment (sauloperez) [inline spec/features/admin/order_cycles/simple_spec.rb:34]: IMO this particular comment that we repeat doesn't add much value. I think Capybara's DSL already conveys that.

File: spec/features/admin/order_cycles/simple_spec.rb [hunk 4]
```diff
@@ -12,53 +12,58 @@
...
+    within(".flatpickr-calendar.open") do
+      # Then select first of month 
+     find('.dayContainer .flatpickr-day', text: "1").click
+   end
+
     within("tr.order-cycle-#{oc2.id}") do
-      find("input#oc#{oc2.id}_name").set "Updated Order Cycle 2"
-      find("input#oc#{oc2.id}_orders_open_at").set "2040-12-01 12:00:02"
-      find("input#oc#{oc2.id}_orders_close_at").set "2040-12-01 12:00:03"
+      # Then that date/time should appear on the form
```
Comment (sauloperez) [inline spec/features/admin/order_cycles/simple_spec.rb:47]: this one as well

File: spec/features/admin/order_cycles/simple_spec.rb [hunk 5]
```diff
@@ -12,53 +12,58 @@
...
+    ## -- OC1
+    # Manually fill out name
+    find("input#oc#{oc1.id}_name").set ""
+    fill_in("oc#{oc1.id}_name", :with => "Updated Order Cycle 1")
+    
+    ## -- OC2
+    # Manually fill out name
+    fill_in("oc#{oc2.id}_name", :with => "Updated Order Cycle 2")
+    within("tr.order-cycle-#{oc2.id} .orders_open_at") do
+      # When I trigger the datepicker
```
Comment (sauloperez) [inline spec/features/admin/order_cycles/simple_spec.rb:37]: that feels a bit redundant as well

File: app/assets/javascripts/admin/all.js [hunk 6]
```diff
@@ -26,6 +25,23 @@
...
+//= require flatpickr/l10n/es
+//= require flatpickr/l10n/fr
+//= require flatpickr/l10n/it
+//= require flatpickr/l10n/nl
+//= require flatpickr/l10n/pl
+//= require flatpickr/l10n/pt
+//= require flatpickr/l10n/ru
+//= require flatpickr/l10n/sv
+//= require flatpickr/l10n/tr
+//= require shortcut-buttons-flatpickr.min
```
Comment (sauloperez) [inline app/assets/javascripts/admin/all.js:43]: managing this dependency without a package manager scares me a bit and it's something we've been trying to overcome. 

Rails' historical mechanism it's been to provide gems ready to be used by the asset pipeline such as https://github.com/adambutler/vuejs-rails or https://github.com/rails/jquery-rails. They keep up to date with the JS dependency so we only need to deal with updating the gem.

Including the JS file as-is, this dependency won't be updated until it's too late. That's why we introduced yarn some time ago plugged it into Rails' asset pipeline (as Rails does with its webpack support in future versions). You can take https://github.com/openfoodfoundation/openfoodnetwork/pull/6049 as an example. The goal in the long-term is to have 0 vendorized JS dependencies.

Perhaps @luisramos0 can shed some more light into this.

**Review by sauloperez — CHANGES_REQUESTED** [review]

**Comment by sauloperez** [comment]
This is some serious effort @jibees ! congrats :clap: :muscle:

File: config/locales/en.yml (hunk 1 above)
Comment (jibees) [reply config/locales/en.yml:3588]: Yes. How can we be sure of that? Is this something I need to check in the future? What should I do?

File: spec/features/admin/bulk_order_management_spec.rb (hunk 2 above)
Comment (jibees) [reply spec/features/admin/bulk_order_management_spec.rb:472]: It's a tag that allow the tester to test only tests tagged with this. 
https://relishapp.com/rspec/rspec-core/v/2-4/docs/command-line/tag-option
And I forgot to delete this one... Thanks!

File: spec/features/admin/order_cycles/simple_spec.rb (hunk 3 above)
Comment (jibees) [reply spec/features/admin/order_cycles/simple_spec.rb:34]: Agree.

File: app/assets/javascripts/admin/all.js (hunk 6 above)
Comment (jibees) [reply app/assets/javascripts/admin/all.js:43]: I totally agree and I missed it. Will make the correction! Thanks for pointing that out to me!

File: config/locales/en.yml (hunk 1 above)
Comment (sauloperez) [reply config/locales/en.yml:3588]: we should let the next release manager know so we wait for translators and include their translations in the release.

File: app/assets/javascripts/admin/all.js (hunk 6 above)
Comment (jibees) [reply app/assets/javascripts/admin/all.js]: Commit 6d07fc4 should resolve. Let me know.

File: config/locales/en.yml (hunk 1 above)
Comment (jibees) [reply config/locales/en.yml:3588]: The next release manager is @Matt-Yorkley. So, I let you know Matt!

File: yarn.lock [hunk 7]
```diff
@@ -1514,6 +1514,11 @@ setprototypeof@1.1.1:
   resolved "https://registry.yarnpkg.com/setprototypeof/-/setprototypeof-1.1.1.tgz#7e95acb24aa92f5885e0abef5ba131330d4ae683"
   integrity sha512-JvdAWfbXeIGaZ9cILp38HntZSFSo3mWg6xGcJJsd+d4aRMOqauag1C63dJfDw7OaMYwEbHMOxEZ1lqVRYP2OAw==
 
+shortcut-buttons-flatpickr@^0.3.1:
+  version "0.3.1"
+  resolved "https://registry.yarnpkg.com/shortcut-buttons-flatpickr/-/shortcut-buttons-flatpickr-0.3.1.tgz#25ec705db8e7c80f0f81f5e160783102775ef529"
+  integrity sha512-zZRLJXF7WSeQJ0ZES+PdD1FI+yjBOufqOQ4AU5przR71Rhyekiw/6/JLA33zOqVkgR7QQ1NbVcuuSqNHHGneEg==
+
```
Comment (sauloperez) [inline yarn.lock:1521]: ah so, it doesn't directly depend on the flatpickr package. I guess then it's very unlikely we'll run into issues due to having `flatpickr` managed by Bundler and this plugin managed by Yarn.

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
By the look of the build's output it seems you need to rebase onto `master`.

File: yarn.lock (hunk 7 above)
Comment (jibees) [reply yarn.lock:1521]: You're right: let's put `flatpickr` in Yarn.

Comment (jibees) [reply yarn.lock:1526]: Commit 42f6ed0 should resolve this. Thanks Pau!

**Comment by jibees** [comment]
> And it looks CodeClimate's scss-lint requires us to exclude some things but fix some others.
Which things are you thinking of? Sometimes I need to override some css selectors provided by flatpickr itself so I'm not decision-maker on this (for exemple, each time you see `Selector flatpickr-monthDropdown-months should be written in lowercase with hyphens` or `Avoid qualifying class selectors with an element.` this is because I need to override a previous selector ) ;

**Comment by sauloperez** [comment]
I just did a quick scan and noticed that maybe some apply. That's why I would add the necessary in-line comment so scss-lint excludes those that not under our control. If that gets too dirty we can perhaps exclude these rules for this particular file in .codeclimate.yml. I don't know if that possible, sorry, just thinking out loud.

File: Gemfile_next.lock [hunk 8]
```diff
@@ -286,9 +285,9 @@ GEM
     mini_racer (0.2.15)
       libv8 (> 7.3)
     minitest (5.14.2)
-    monetize (1.9.4)
+    monetize (1.10.0)
```
Comment (sauloperez) [inline Gemfile_next.lock:288]: I checked Gemfile_next.lock in `master` and the versions don't match. It'd be safer to rebase onto `master`.

File: Gemfile_next.lock [hunk 9]
```diff
@@ -320,11 +319,11 @@ GEM
       activerecord (>= 4.0, < 6.1)
     parser (2.7.2.0)
       ast (~> 2.4.1)
-    paypal-sdk-core (0.2.10)
+    paypal-sdk-core (0.3.4)
       multi_json (~> 1.0)
       xml-simple
-    paypal-sdk-merchant (1.106.1)
-      paypal-sdk-core (~> 0.2.3)
+    paypal-sdk-merchant (1.117.2)
```
Comment (sauloperez) [inline Gemfile_next.lock:325]: This one as well. This is the version we have in Gemfile.lock but not in Gemfile_next.lock

**Review by sauloperez — CHANGES_REQUESTED** [review]

**Comment by jibees** [comment]
> I just did a quick scan and noticed that maybe some apply. That's why I would add the necessary in-line comment so scss-lint excludes those that not under our control. If that gets too dirty we can perhaps exclude these rules for this particular file in .codeclimate.yml. I don't know if that possible, sorry, just thinking out loud.

AFAIK, I can exclude some files (as explain here: https://docs.codeclimate.com/docs/excluding-files-and-folders) ; but i didn't find anything that let me add instruction into the file `flatpickr-customization.scss` to disable check.

On the other hand, it seems to be possible through the application like this way :
Capture d’écran 2021-01-28 à 15 26 58

What do you thinks?

**Comment by sauloperez** [comment]
but that's a valid rule that we'll want to apply most of the time, right? that will disable it forever AFAIK.

File: Gemfile [hunk 10]
```diff
@@ -141,6 +141,7 @@ group :production, :staging do
   gem 'unicorn-worker-killer'
 end
 
+
```
Comment (mkllnk) [inline Gemfile:144]: Why did you add this line?

**Review by mkllnk — COMMENTED** [review]
This is looking good. It needs a bit of a clean-up though. I'm moving it back to in-dev until the Gemfile issues are addressed.

File: Gemfile (hunk 10 above)
Comment (jibees) [reply Gemfile:144]: Actually I added a line with content, then deleted the content but without the line itself. Will delete the line.

**Comment by jibees** [comment]
> but that's a valid rule that we'll want to apply most of the time, right? that will disable it forever AFAIK.
@sauloperez 
I added `// scss-lint:disable QualifyingElement` and `// scss-lint:disable SelectorFormat` on top of concerned files. See commit a96e163 and ff4fdf4

File: Gemfile_next.lock (hunk 8 above)
Comment (jibees) [reply Gemfile_next.lock]: Should be resolved by commit f8e7b76caf50a53810bbbc3700d274be97bbcdf3

File: Gemfile_next.lock (hunk 9 above)
Comment (jibees) [reply Gemfile_next.lock]: Should be resolved by commit f8e7b76caf50a53810bbbc3700d274be97bbcdf3

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
This is great @jibees ! the git history is much better now :clap: 

Just in case, if you haven't read it yet, https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html is to me the bible on this topic.

**Comment by filipefurtad0** [comment]
Hey @jibees ,

There was an error which prevented the deployment of this PR on staging-FR, see:
https://app.bugsnag.com/open-food-france/coopcircuits/errors/60108edbe7105a001869ffaa?event_id=601be5cd0069b53aea960000&i=sk&m=nw

This appeared on devops-notifications channel. Adding the feedback-needed label.

**Comment by jibees** [comment]
Thanks for pointing me that out. This should now be resolved.

**Review by mkllnk — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @jibees ,

This looks really nice! :clap: :clap: :clap: 

I had a look at these 5 pages on **desktop**, and splitted them into `date-time-pickers` and `date-pickers`:

`date-time-pickers`:

-  **/admin/order_cycles/** - OK

image

-  **/admin/order_cycles/[ORDER_ID]/edit** - OK

image


- **Reports** - checked the following - all OK

Orders And Distributors
Bulk Co-Op
Payment Reports
Orders & Fulfillment Reports
Enterprise Fee Summary
Order Cycle Management
Sales Tax
Xero Invoices 
Packing Reports

image

`date-pickers:`

-    **/admin/orders/bulk_management**

image

-    **/admin/orders?q[s]=completed_at+desc#**

image

- checked subscriptions settings page - **/admin/subscriptions/[SUBS_ID]/**edit - as well:

image


Going through the acceptance criteria:

i) (must) Select both date and time -> I've found date-pickers (ex., for order management) and date-time-pickers (ex., for order cycles) to be displayed on the same pages as before. Is this really intended for **all** pages, as described in #6664? 

ii) Is or already or can be translated - I think we're good here, I've noticed you've introduced the missing key for "Now" and "Today" :+1:
 
iii) Allow the option in the future to add custom shortcuts such “next month” (as discussed here) - I'm not sure I can verify this here, but indeed there is plenty of space left, on the bottom of the calendar layout, so I think this should be possible to add.

iv) Currently the date picker has the option to select NOW/TODAY. It could be nice to keep it. - still there, all good. It even introduces the "Today" option for "date-pickers", which is not the available in production :+1:  

v) Improvement on the selection of the end date. I noticed that the default hour is selected when clicking a given day is 12:00. So if one is aiming at selecting the 04.02, clicking on "Start at" and "End at" will not work:

image

So I think the default "Start at" time should be at 00:00, and the "End at" rather at 23:59, to fully include that day? I think this way we are sure the clicked day is fully selected. 


I would say this is ready to go, despite of points i) and v), but I have some additional comments regarding testing in mobile. This relates to:

vi) the "Done" button and/or the closing of the calendar (upon clicking)

The previous calendar layout had a "Done" button, which was removed on this new layout. I think this was intentional (as the "Done" key was removed from `en.yml`) and comparing with the examples in https://flatpickr.js.org/examples/ tells me this is indeed the intended behavior (As a "OFN-user", I kind of missed it, but I think I'd quickly get used to it). Anyway, and having this in mind, I've found that `date-pickers` behave differently than `date-time-pickers`:

- `date-pickers`: if you click on a day, the calendar accepts your choice and closes
- `date-time-pickers`: if you click on a day, the calendar does not close - to close it you'll need to close it by clicking somewhere else on the screen.

While at first, this seems a minor usability issue, it shows significant differences on **mobile**, especially on Firefox. I believe this is a feature if this specific calendar, from flatpickr. Specifically, and using **Firefox** on **mobile devices**, does not really allow to close the **date-time-pickers** layout, as it takes the whole page, on these devices. This is not an issue with this PR, because it is reproducible on the **flatpickr** examples page. See the videos below:

**Firefox, on Android 10 Samsung Galaxy S20 (and also reproduced on my own phone, a Huawei P8 Lite):**

Peek 2021-02-09 22-44

See the page https://flatpickr.js.org/examples/

image

This issue is not observed on **Android using Chrome**:

Peek 2021-02-09 22-48_chrome


Nor on **iPhone 12/Safari**:

Peek 2021-02-09 22-52_iphone


So, in summary this PR:
i) keeps date/time pickers on the same pages as before (I think this is the correct and good to go, unless we want to introduce day and time, on all calendars)

v) full day selection could be improved, as default time is 12:00  

vi) seems to raise serious usability issues _only_ to users browsing with Firefox on mobile devices, but in these cases it basically prevents the user to select a date/hour. (the dirty workaround is to rotate the phone 90°, not sure this is acceptable).

Thoughts @jibees @RachL @Erioldoesdesign ?

**Comment by jibees** [comment]
Hi Filipe!
What a piece of work, thanks a lot for that! 🙌 🙏
My answers: 
 - Yes, we (maybe I?) chose to keep the same behavior as before: only date when this was only date, date & time when it was date and time.
 - If I undestand correctly (please correct me if needed), this is the same behavior as today (by default the selecting date is today at 00:00). Saying that, I totally agree that this behavior could be improved!
Capture d’écran 2021-02-10 à 09 26 17


 - For the mobile version, flatpickr does use this one code, but call the native date(time)picker. So, this is not a flatpickr bug, but a Android/Firefox bug. Actually, this is a serious bug for Firefox, and this bug will appear each time an app calls this native datetimepicker. 
 - Yes, the done button is considered as unnecessary using this new UX. This is how flatpickr handles it. If its a big lack, I imagine I could _easily_ add a "Done" button, but we must be aware of UX implication is this case: when no date is selected, a date bu no time...

**Comment by filipefurtad0** [comment]
Hey @jibees thanks for the quick reply!

Replying to the points above:
> only date when this was only date, date & time when it was date and time

Yes, this sounds good. Anyway I think it should not hold this PR from being merged, we can deal with it in a separate issue, if it is indeed a requirement.

> this is the same behavior as today (by default the selecting date is today at 00:00)

I see this now, thanks for pointing this out. In some cases, for example `/admin/reports/orders_and_fulfillment` in which today's date is pre-set by default on the form, it indeed remains unchanged, and after this PR:

image

However in other pages and use cases in which no default date is set - for example: `/admin/reports/orders_and_distributors`, or after cloning an OC in which the date/time fields are empty - this PR introduces a change in the default time set by the calendar: before it introduced 00:00 and now introduces 12:00:

image

But I do see your point: It does not change the automatic selection of a whole day, so improvements can be addressed on a separate PR. I'll open an issue for this.

> this is not a flatpickr bug, but a Android/Firefox bug. Actually, this is a serious bug for Firefox

Cool, thanks for that info! I don't particularly advocate for keeping "Done" button. However, we should be aware that **admin users** on **mobile + Firefox** won't be able to use date-time-picker calendars. Also, and considering that this PR greatly simplifies the UI (which is great!) it does remove the possibility to manually type on these date/time fields, and as such it will render certain features blocked (like editing the opening of order cycle), for these users. So I think merging as is, is a product's @openfoodfoundation/train-drivers-product-owners decision :+1: 

Moving to ready to go , but adding the Feedback label.

**Comment by RachL** [comment]
@jibees @filipefurtad0 Haven't had time to read everything in detail, so forget my comment if completely irrelevant, but:

> this is the same behavior as today (by default the selecting date is today at 00:00)

this is not true on pages like `/admin/orders?q[s]=completed_at+desc` when you select an end date let's say February the 8th, it does include February the 8th in the search. this was specifically changed a few month ago. Just flagging that we cannot introduce a regression here. However in other cases, if it does not change the behavior then yes these type of criteria in the original issue were more a "nice to have" :)

As for mobile and admin view... This was not part of the criteria. Should it have been? I'm really wondering. In any case I don't think it should block the merge. We can always work on the mobile aspect in a future / bigger epic on mobile or separately once our goals for mobile use in the backend are clear. I assume that IF at the end we need to choose something else than flatpickr, this PR now makes it easy to change anyway? 

Also @jibees do you have "proof" that this is really a Firefox bug? I see a log of Firefox issues in their repo, but most are taking care of by their team it seems?

**Comment by filipefurtad0** [comment]
Thanks for feedback @RachL ,

That's very relevant, of course:
> February the 8th, it does include February the 8th in the search

I've just checked that case (the case in which `date-pickers` are used), and I think it is covered in this PR. For an order placed today (10 Feb.), selecting the 10th of February on the calendar includes this order in the results - so no regression here, so as I understand, no additional issue is necessary here:

image

As to the second point raised:
> In any case I don't think it should block the merge.

Ok, thanks! In that case, I'll add an issue on the `date-time-picker` on mobile/Firefox as this is a regression but, as mentioned, only for a restrict group. Moving to ready to go! :+1:

**Comment by jibees** [comment]
> Also @jibees do you have "proof" that this is really a Firefox bug? I see a log of Firefox issues in their repo, but most are taking care of by their team it seems?

I should have a better explanation. It's not as wide as a Android/Firefox bug. It's maybe a bit trickier. 
I've tested on my side, and works as expected:
Screenshot_2021-02-10-21-14-35-09
(NB: you could see the padding/margin issue on top-right side ;))
To confirm this is not on our side, @filipefurtad0, could you test with your mobile this site (as you noticed you had a bug with your Huawei P8 Lite), which use the native date time picker : https://developer.mozilla.org/fr/docs/Web/HTML/Element/Input/datetime-local (it works for me as well)

**Comment by jibees** [comment]
I've tested on BrowserStack with your configuration @filipefurtad0 (Firefox/Galaxy 20) and with a website unrelated to OFN nor flatpickr. I still have a bug : 
Capture d’écran 2021-02-11 à 10 28 15
I'm now 100% sure it's not on our side.

**Comment by filipefurtad0** [comment]
Thanks for double checking @jibees ! I've opened issue #6859 - yes I agree, it's not on our side :+1:
